src/Service/Resolvers/SqlQueryEngine.cs

```diff
@@ -213,6 +214,39 @@ private OkObjectResult FormatFindResult(JsonDocument jsonDoc, FindRequestContext
             // nextLink is the URL needed to get the next page of records using the same query options
             // with $after base64 encoded for opaqueness
             string path = UriHelper.GetEncodedUrl(_httpContextAccessor.HttpContext!.Request).Split('?')[0];
+
+            // If the base route is not empty, we need to insert it into the URI before the rest path.
+            string baseRoute = _runtimeConfigProvider.RestBaseRoute;
```

Comment:
With #1402 we'll go `runtimeConfig.Runtime.Rest.BaseRoute` rather than a shadowed property off the provider.